### DISCUSSION TITLE
Menu triggered by right click

### DISCUSSION THREAD

**Discussion by Enuns** [mantinedev/mantine#7040] [Feature requests]
I'd like to request an enhancement for the Menu component in Mantine to support a right-click (context menu) trigger. This would enable a more intuitive user experience, particularly for applications that require context-specific actions triggered by right-clicks, such as dashboards, file management interfaces, or any setting where additional menu options are relevant to specific content areas.

I know there is an alternative (https://icflorescu.github.io/mantine-contextmenu), but I got to change the whole structure that I've already built just to change the trigger event

**Comment by Kenzo-Wada** [comment]
radix ui and others offer this component and I thought it was a great idea.
https://www.radix-ui.com/themes/docs/components/context-menu

@rtivital 
What do you think? Also, I can use my time to implement it on my end :)

**Comment by minosss** [comment]
The mantine's popup component is based on floating-ui. You can wrap a new trigger. 

I do this. https://gist.github.com/minosss/f26fae6170d62df26103a0c589bf6da6

use likes Menu

```tsx
<ContextMenu position='top-end'>
  <ContextMenu.Target>
    <Center h={100} bg='teal'>
      Right Click
    </Center>
  </ContextMenu.Target>
  <ContextMenu.Dropdown>
    <ContextMenu.Item>Undo</ContextMenu.Item>
    <ContextMenu.Item>Redo</ContextMenu.Item>
  </ContextMenu.Dropdown>
</ContextMenu>
```

**Comment by IsaacInsoll** [comment]
I'd also love to see this functionality supported natively by Mantine.
